### PR TITLE
Add MusicXML sound tag parsing (fix #1578)

### LINKED ISSUE #1578: part.write("midi", ) output different tempo MIDI file.
**music21 version**

`8.1.0`   

**Problem summary**

Parse `.musicxml` file to music21 stream, then output parts to MIDI file, only the first part have the correct tempo.

**Steps to reproduce**
Files for reproduce:
[sample.zip](https://github.com/cuthbertLab/music21/files/11498226/sample.zip)

``` python
from music21 import *

score = converter.parse("sample.musicxml")

for p in score.parts:
    p.atSoundingPitch = False
    for m in p.getElementsByClass("Measure"):
        m.removeByClass("ChordSymbol")

piano_rh = score.parts[0]
piano_rh.write("midi","sample_rh.mid")
piano_lh = score.parts[1]
piano_lh.write("midi","sample_lh.mid")
```

**Expected vs. actual behavior**

Expected behavior:
Both MIDI files have tempo = 40.

Actual behavior:
Only the first part output with the correct tempo.

image
image

**More information**

I don't know if this is expected behavior. Because as I know "Part" object does not contain tempo information.

### PR DESCRIPTION
**Pull request by TimFelixBeyer**
This PR adds functionality to parse `<sound tempo='x'>` tags from MusicXML files into MetronomeMark objects (fixing #1578). 
`bach/bwv66.6` contains such a `<sound tempo='x'>` tag and is heavily used in doctests. Therefore a lot of tests had to be updated, leading to quite a lot of changes, however nothing seems to be broken. 
The only tests that had significantly different outputs are related to feature extraction.
They change because the changed tempo leads to a (very) different beat histogram. 
Previously, the tempo was 120bpm, leading to beat bpm values of 60, 120, 240, where 240 was discarded due to the this line: https://github.com/cuthbertLab/music21/blob/acf926c64215660467ca7de8b05e5bb50dfb6a27/music21/features/base.py#L504 
The accurate tempo is 96, so the values are 48, 96, 192 and are therefore no longer discarded.

Let me know what you think!

### CODE REVIEW

File: music21/musicxml/xmlToM21.py [hunk 1]
```diff
@@ -5312,6 +5324,53 @@ def setDirectionInDirectionType(self, mxDir, mxDirection, staffKey, totalOffset)
...
+        totalOffset = offsetDirection + self.offsetMeasureNote
+
+        staffKey = self.getStaffNumber(mxSound)
+
+        self.setSound(mxSound,
+                      None,
+                      staffKey,
+                      totalOffset)
+
+    def setSound(self, mxSound, mxDir, staffKey, totalOffset):
```
[the remark this answers is not in the document]
Comment (mscuthbert) [reply music21/musicxml/xmlToM21.py:5343]: add at least some documentation so that it appears in docs with some idea of what it does.

**Review by mscuthbert — APPROVED** [review]
Approved!  This is a huge improvement.  Thank you!  I had no idea there were so many XML files with tempos specified in `<sound>` elements but without metronomes.  I'm going to close and re-open to see if there are any conflicts with the stripTies bug fix.

**Comment by mscuthbert** [comment]
Indeed - - the typing did catch a minor bug and added a warning that something that appeared typed was not.  Thanks for being totally on top of this!

**Comment by TimFelixBeyer** [comment]
A file in the corpus (schoenberg/opus19) had an incorrect <sound tempo=0> tag, I have removed it.

**Comment by mscuthbert** [comment]
It also just looks more musical to see qtr=144 than 144.0, and
more accurate to the composer's notation. So good!

**Comment by mscuthbert** [comment]
Sorry for the delay in merging -- I just needed to check out the Schoenberg file and make sure all was good.

**Comment by mscuthbert** [comment]
I'm changing this a little bit in the next PR because I'm seeing phantom metronome marks in lots of pieces that should not show them in scores (like medieval/Renaissance music, Bach Chorales, etc.) -- these only set numberSounding so that the metronome mark does not appear in the score, just affecting playback.

**Pull request merged**